### PR TITLE
fix: Allow date handling for feature flags

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/models/property/property.py [hunk 1]
```diff
@@ -318,6 +318,11 @@ def property_to_Q(self) -> Q:
                 | Q(**{f"{column}__{self.key}": None})
             )
 
+        # TODO: add a test for this
```
Comment (neilkakkar) [inline posthog/models/property/property.py:321]:
```suggestion

```

File: posthog/models/team/team.py [hunk 2]
```diff
@@ -208,8 +209,13 @@ def actor_on_events_querying_enabled(self) -> bool:
             return posthoganalytics.feature_enabled(
                 "person-on-events-enabled",
                 str(self.uuid),
-                groups={"project": str(self.uuid)},
-                group_properties={"project": {"id": str(self.pk)}},
+                groups={"organization": str(self.organization.id)},
+                group_properties={
+                    "organization": {
+                        "id": str(self.organization.id),
+                        "created_at": self.organization.created_at.replace(tzinfo=timezone.utc),
```
Comment (neilkakkar) [inline posthog/models/team/team.py:216]:
```suggestion
                        "created_at": self.organization.created_at,
```

File: posthog/test/test_team.py [hunk 3]
```diff
@@ -75,8 +76,13 @@ def test_team_on_cloud_uses_feature_flag_to_determine_person_on_events(self, moc
                 mock_feature_enabled.assert_called_once_with(
                     "person-on-events-enabled",
                     str(team.uuid),
-                    groups={"project": str(team.uuid)},
-                    group_properties={"project": {"id": str(team.pk)}},
+                    groups={"organization": str(self.organization.id)},
+                    group_properties={
+                        "organization": {
+                            "id": str(self.organization.id),
+                            "created_at": self.organization.created_at.replace(tzinfo=timezone.utc),
```
Comment (neilkakkar) [inline posthog/test/test_team.py:83]:
```suggestion
                            "created_at": self.organization.created_at,
```

File: requirements.in [hunk 4]
```diff
@@ -47,7 +47,7 @@ numpy==1.21.4
 parso==0.8.1
 pexpect==4.7.0
 pickleshare==0.7.5
-posthoganalytics==2.1.0
+posthoganalytics==2.1.1
```
Comment (neilkakkar) [inline requirements.in:50]:
```suggestion
posthoganalytics==2.1.2
```

File: requirements.in [hunk 5]
```diff
@@ -47,7 +47,7 @@ numpy==1.21.4
 parso==0.8.1
 pexpect==4.7.0
 pickleshare==0.7.5
-posthoganalytics==2.1.0
```
Comment (EDsCODE) [inline requirements.in:50]: Was just about to call this out

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**